cmd/dp-dimension-importer/main.go

```diff
@@ -25,61 +42,123 @@ func (r responseBodyReader) Read(reader io.Reader) ([]byte, error) {
+
+	signals := make(chan os.Signal)
+	signal.Notify(signals, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+
 	cfg, err := config.Load()
 	if err != nil {
-		log.ErrorC("Error while attempting to load application config.", err, nil)
+		log.ErrorC(loadConfigErr, err, nil)
 		os.Exit(1)
 	}
 
 	log.Debug("application configuration", log.Data{"config": cfg})
 
-	extractedEventConsumer, err := kafka.NewConsumerGroup(cfg.KafkaAddr, cfg.DimensionsExtractedTopic, log.Namespace, kafka.OffsetNewest)
-	if err != nil {
-		log.ErrorC("could not create consumer", err, nil)
-		os.Exit(1)
-	}
+	// Incoming kafka topic for instances to process
+	instanceConsumer := newConsumer(cfg.KafkaAddr, cfg.IncomingInstancesTopic, log.Namespace)
 
-	insertedEventProducer, err := kafka.NewProducer(cfg.KafkaAddr, cfg.DimensionsInsertedTopic, 0)
-	if err != nil {
-		log.ErrorC("kafka producer error", err, log.Data{"topic": cfg.DimensionsInsertedTopic})
-		os.Exit(1)
-	}
+	// Outgoing topic for instances that have completed processing
+	instanceCompleteProducer := newProducer(cfg.KafkaAddr, cfg.OutgoingInstancesTopic)
 
-	var neo4jClient *client.Neo4j
-	if neo4jClient, err = client.NewNeo4j(cfg.DatabaseURL, cfg.PoolSize); err != nil {
-		log.ErrorC("unexpected error while to create database connection pool", err, log.Data{
-			logKeys.URL:      cfg.DatabaseURL,
-			logKeys.PoolSize: cfg.PoolSize,
-		})
-		os.Exit(1)
-	}
+	// Outgoing topic for any errors while processing an instance
+	errorEventProducer := newProducer(cfg.KafkaAddr, cfg.EventReporterTopic)
+
+	neo4jClient := createNeo4jClient(cfg)
 
 	newDimensionInserterFunc := func() handler.DimensionRepository {
 		return repository.NewDimensionRepository(neo4jClient, map[string]string{})
 	}
 
-	datasetAPI := client.DatasetAPI{
-		ResponseBodyReader:  responseBodyReader{},
-		HTTPClient:          &http.Client{},
-		DatasetAPIHost:      cfg.DatasetAPIAddr,
-		DatasetAPIAuthToken: cfg.DatasetAPIAuthToken,
+	// ImportAPI HTTP client.
+	datasetAPICli := client.NewDatasetAPI(cfg.DatasetAPIAddr, cfg.DatasetAPIAuthToken, responseBodyReader{}, &http.Client{})
+
+	// Handler for dimensionsExtracted events.
+	eventHandler := handler.NewDimensionExtractedEventHandler(
+		newDimensionInserterFunc,
+		&repository.InstanceRepository{Neo4j: neo4jClient},
+		datasetAPICli)
+
+	// MessageProducer for dimensionsInsertedEvents
+	instanceCompletedProducer := message.NewInstanceCompletedProducer(
+		instanceCompleteProducer,
+		schema.InstanceCompletedSchema,
+	)
+
+	// Errors handler
+	errorEventHandler := &handler.ErrorHandler{
+		Producer:   errorEventProducer,
+		Marshaller: schema.ErrorEventSchema,
 	}
 
-	eventHandler := &handler.DimensionsExtractedEventHandler{
-		NewDimensionInserter: newDimensionInserterFunc,
-		InstanceRepository:   &repository.InstanceRepository{Neo4j: neo4jClient},
-		DatasetAPI:           datasetAPI,
+	// HTTP Health check endpoint.
+	healthcheck.NewHandler(cfg.BindAddr, cfg.ShutdownTimeout)
```

Comment:
You don't need to pass in a shutdown timeout to set on the http server. As you are passing a context into the close function of the healthcheck it will use the timeout on that.